discovery/db_discovery.go

```diff
@@ -145,6 +149,53 @@ func (dbo *DBOrchestratorPoolCache) cacheTranscoderPool() error {
 	return nil
 }
 
+func (dbo *DBOrchestratorPoolCache) cacheOrchestratorStake() error {
+	orchs, err := dbo.store.SelectOrchs(
+		&common.DBOrchFilter{
+			CurrentRound: dbo.rm.LastInitializedRound(),
+		},
+	)
+	if err != nil {
+		return fmt.Errorf("could not retrieve orchestrators from DB: %v", err)
+	}
+
+	resc, errc := make(chan *common.DBOrch), make(chan error)
+	ctx, cancel := context.WithTimeout(context.Background(), getOrchestratorsTimeoutLoop)
+	defer cancel()
+
+	currentRound := dbo.rm.LastInitializedRound()
+
+	getStake := func(o *common.DBOrch) {
+		ep, err := dbo.lpEth.GetTranscoderEarningsPoolForRound(ethcommon.HexToAddress(o.EthereumAddr), currentRound)
+		if err != nil {
+			errc <- err
+			return
+		}
+		o.Stake = ep.TotalStake.String()
+		resc <- o
+	}
+
+	for _, o := range orchs {
+		go getStake(o)
```

Comment:
Creating a goroutine for each query feels like a premature optimization at this point IMO since it introduces a piece of code that needs to be concurrency safe (and ideally we would test for it). Caching the stake for each active O isn't that time sensitive since it just happens on startup. Additionally, a simpler optimization that we could use in the future is to write a wrapper contract that aggregates all the stake values for a list of Os so that we can fetch the results with a single query.

Reply:
If I'm not mistaken go-sqlite does use a RWmutex so this should be thread safe regardless, since we're only writing the value to the DB and not memory.

Reply:
Yeah good point. On second thought, I'm fine with the concurrent requests used here 👍 with the wrapper contract as a nice to have optimization later on that removes the need for any concurrency code.